internal/lsp/server_template_test.go

```diff
@@ -124,6 +116,73 @@ func TestTemplateContentsForFile(t *testing.T) {
 	}
 }
 
+func TestTemplateContentsForFileInWorkspaceRoot(t *testing.T) {
+	t.Parallel()
+
+	td := t.TempDir()
+
+	err := os.MkdirAll(filepath.Join(td, ".regal"), 0o755)
+	if err != nil {
+		t.Fatalf("failed to create directory %s: %s", filepath.Join(td, ".regal"), err)
+	}
+
+	err = os.WriteFile(filepath.Join(td, ".regal/config.yaml"), []byte{}, 0o600)
+	if err != nil {
+		t.Fatalf("failed to create directory %s: %s", filepath.Join(td, ".regal"), err)
+	}
+
+	ctx := context.Background()
+
+	s := NewLanguageServer(ctx, &LanguageServerOptions{ErrorLog: newTestLogger(t)})
+	s.workspaceRootURI = uri.FromPath(clients.IdentifierGeneric, td)
+
+	fileURI := uri.FromPath(clients.IdentifierGeneric, filepath.Join(td, "foo.rego"))
+
+	s.cache.SetFileContents(fileURI, "")
+
+	_, err = s.templateContentsForFile(fileURI)
+	if err == nil {
+		t.Fatalf("expected error")
+	}
+
+	if !errors.Is(err, &templatingInRootError{}) {
+		t.Fatalf("expected error to be templatingInRootError, got %T", err)
+	}
+}
+
+func TestTemplateContentsForFileWithUnknownRoot(t *testing.T) {
```

Comment:
When the root is unknown, but there is a path, we can attempt to use the server's root URI as a root instead.